File: src/NodeParser/NumberTypeNodeParser.ts
import ts from "typescript";
import type { Context } from "../NodeParser.js";
import type { SubNodeParser } from "../SubNodeParser.js";
import type { BaseType } from "../Type/BaseType.js";
import { NumberType } from "../Type/NumberType.js";

export class NumberTypeNodeParser implements SubNodeParser {
    public supportsNode(node: ts.KeywordTypeNode): boolean {
        return node.kind === ts.SyntaxKind.NumberKeyword || node.kind === ts.SyntaxKind.BigIntKeyword;
    }
    public createType(node: ts.KeywordTypeNode, context: Context): BaseType {
        return new NumberType();
    }
}

File: internal/parser/number_type_node_parser.go
package parser

import (
	"github.com/microsoft/typescript-go/shim/ast"

	"github.com/vega/ts-json-schema-generator/internal/types"
)

// NumberTypeNodeParser handles the `number` and `bigint` keywords
// (src/NodeParser/NumberTypeNodeParser.ts).
type NumberTypeNodeParser struct{}

func NewNumberTypeNodeParser() *NumberTypeNodeParser {
	return &NumberTypeNodeParser{}
}

func (p *NumberTypeNodeParser) SupportsNode(node *ast.Node) bool {
	return node.Kind == ast.KindNumberKeyword || node.Kind == ast.KindBigIntKeyword
}

func (p *NumberTypeNodeParser) CreateType(_ *ast.Node, _ *Context, _ *types.ReferenceType) types.Type {
	return &types.NumberType{}
}
